build_sis_analyst_multi_turn_extended.py

```diff
@@ -0,0 +1,308 @@
+"""
+Cortex Analyst App
+====================
+This app allows users to interact with their data using natural language.
+"""
+
+import json  # To handle JSON data
+import time
+from typing import Dict, List, Optional, Tuple
+
+import _snowflake  # For interacting with Snowflake-specific APIs
+import pandas as pd
+import streamlit as st  # Streamlit library for building the web app
+from snowflake.snowpark.context import (
+    get_active_session,
+)  # To interact with Snowflake sessions
+from snowflake.snowpark.exceptions import SnowparkSQLException
+
+# List of available semantic model paths in the format: <DATABASE>.<SCHEMA>.<STAGE>/<FILE-NAME>
+# Each path points to a YAML file defining a semantic model
+AVAILABLE_SEMANTIC_MODELS_PATHS = [
+    "CORTEX_ANALYST_DEMO.REVENUE_TIMESERIES.RAW_DATA/revenue_timeseries.yaml"
+]
+API_ENDPOINT = "/api/v2/cortex/analyst/message"
+API_TIMEOUT = 30000  # in milliseconds
```

Comment:
We shouldn't early timeout here. The API level timeout is 50s
```suggestion
API_TIMEOUT = 50000  # in milliseconds
```